File: tools/xgm2tool/src/sgdk/xgm2tool/format/DataBank.java
package sgdk.xgm2tool.format;

public class DataBank
{
    final public int id;
    final public byte[] data;

    public DataBank(VGMCommand command)
    {
        super();

        if (!command.isDataBlock())
            throw new IllegalArgumentException(String.format("Incorrect sample data declaration at %6X !",
                    Integer.valueOf(command.getOriginOffset())));

        // id
        id = command.getDataBankId();
        // allocate data
        data = new byte[command.getDataBlockLen()];

        // copy bank data
        System.arraycopy(command.data, 7, data, 0, data.length);
    }

    public byte getSample(int offset)
    {
        return data[offset];
    }
}

File: tools/crates/xgm2tool/src/data_bank.rs
use crate::vgm_command::VGMCommand;

/// Raw data bank extracted from a VGM data block command
#[derive(Clone, Debug)]
pub struct DataBank {
    pub id: u8,
    pub data: Vec<u8>,
}

impl DataBank {
    pub fn new(command: &VGMCommand) -> Self {
        assert!(command.is_data_block(), "Incorrect sample data declaration at {:06X}!", command.origin_offset());
        let id = command.get_data_bank_id();
        let len = command.get_data_block_len();
        let data = command.data[7..7 + len].to_vec();
        DataBank { id, data }
    }

    pub fn get_sample(&self, offset: usize) -> u8 {
        self.data[offset]
    }
}

use crate::command::CommandTrait;
